__tests__/app.test.js

```diff
@@ -202,6 +202,166 @@ describe('GET', () => {
+                .send(inputComment)
+                .then(({body: { msg }}) => {
+                    expect(msg).toBe("Bad Request");
+                });
+            });
+
+            test(`400: When trying to post a comment that has a username which doesnt fit format send a 400: Bad Request`, () => {
+
+                const inputComment = {
+                    username: 2939049,
+                    body: "hello"
+                };
+
+                return request(app)
+                .post('/api/articles/4/comments')
+                .expect(400)
+                .send(inputComment)
+                .then(({body: { msg }}) => {
+                    expect(msg).toBe("Bad Request");
+                });
+            });
+
+            test(`400: When trying to post a comment that has a body which doesnt fit format send a 400: Bad Request`, () => {
+
+                const inputComment = {
+                    username: "icellusedkars",
+                    body: 584857494
+                };
+
+                return request(app)
+                .post('/api/articles/4/comments')
+                .expect(400)
+                .send(inputComment)
+                .then(({body: { msg }}) => {
+                    expect(msg).toBe("Bad Request");
+                });
+            });
+
+            test(`400: When trying to post a comment that is missing username property respond with a 400 bad request`, () => {
+
+                const inputComment = {
+                    body: "I like eggs"
+                };
+
+                return request(app)
+                .post('/api/articles/4/comments')
+                .expect(400)
+                .send(inputComment)
+                .then(({body: { msg }}) => {
+                    expect(msg).toBe("Bad Request");
+                });
+            });
+
+            test(`400: When posted an empty comment object respond with 400: Bad Request`, () => {
+
+                const inputComment = {};
+
+                return request(app)
+                .post('/api/articles/4/comments')
+                .expect(400)
+                .send(inputComment)
+                .then(({body: { msg }}) => {
+                    expect(msg).toBe("Bad Request: comment empty");
+                });
+            });
+
+            test(`404: When trying to post a comment with a username that fits format but doesnt exist in the database respond with 404: username not found`, () => {
+
+                const inputComment = {
+                    username: "not_a_username",
+                    body: "Hey guys whatsup"
+                };
+
+                return request(app)
+                .post('/api/articles/4/comments')
+                .expect(404)
+                .send(inputComment)
+                .then(({body: { msg }}) => {
+                    expect(msg).toBe("Not Found: this author does not exist");
+                });
+            });
+
+        });
+
+    });
+
```

Comment:
Please test if unnecessary properties in the post object are ignored, which means that as long as there is a username and body, the post is successful, no matter how many extra properties are there in the input object.